### PR TITLE
Add (empty) section about running tests to documentation

### PR DESCRIPTION
**Pull request by owen-mc-diffblue**

### CODE REVIEW

File: doc/architectural/compilation-and-development.md [hunk 1]
```diff
@@ -17,6 +17,10 @@ To be documented.
 
 To be documented.
 
+## Running tests
+
+To be documented. (regression tests with test.pl and unit tests with Catch.)
```
Comment (tautschnig) [inline doc/architectural/compilation-and-development.md]: There is (partial) information in doc/architectural/cprover-architecture-overview.md, when discussing the folder `regression`. I think this should be cleaned up in this PR?

Comment (owen-mc-diffblue) [reply doc/architectural/compilation-and-development.md]: Thanks for pointing that out - I hadn't spotted it. I've pushed a new version of this PR which updates the folder walkthrough as well, so we aren't documenting the same thing in two places. I think that documenting how to run the tests is too important to put it in the folder walkthrough, so I've kept it in the Compilation and Development page and put a link to it from the folder walkthrough.

File: doc/architectural/compilation-and-development.md [hunk 2]
```diff
@@ -17,7 +17,29 @@ To be documented.
...
 
-## Documentation
+## Running tests ##
+
+### Regression tests ###
+
+The regression tests are contained in the `regression/` folder.
+They are grouped into directories for each of the tools/modules.
+Each of these contains a directory per test case, containing a C or C++
+file that triggers the bug and a `.desc` file that describes
```
Comment (tautschnig) [inline doc/architectural/compilation-and-development.md]: This needs updating to encompass Java as well as the fact that there can now be multiple tests (and `.desc` files) in a single directory.

File: doc/architectural/folder-walkthrough.md [hunk 3]
```diff
@@ -75,14 +75,10 @@ into the `doc/html` directory when running `doxygen` from `src`.
...
 ## `regression/` ##
 
-The `regression/` directory contains the test suites.
-They are grouped into directories for each of the tools/modules.
-Each of these contains a directory per test case, containing a C or C++
-file that triggers the bug and a `.desc` file that describes
-the tests, expected output and so on. There is a Perl script,
-`test.pl` that is used to invoke the tests as:
+The `regression/` directory contains the regression test suites. How to run the
+regression tests is documented in \ref compilation-and-development
```
Comment (tautschnig) [inline doc/architectural/folder-walkthrough.md]: I would have chosen slightly different wording: "See \ref compilation-and-development for information on how to run and develop regression tests." But then English isn't my first language, so please take with a grain of salt.

**Review by tautschnig — APPROVED** [review]
Nice! Just some comments to be considered at your discretion.

File: doc/architectural/folder-walkthrough.md (hunk 3 above)
Comment (owen-mc-diffblue) [reply doc/architectural/folder-walkthrough.md]: Yes, that is more natural.

File: doc/architectural/compilation-and-development.md (hunk 2 above)
Comment (owen-mc-diffblue) [reply doc/architectural/compilation-and-development.md]: Thanks. I hope this documentation will be updated and expanded soon, but it's worth making sure it isn't factually wrong right now.

**Pull request merged**